CMakeLists.txt

```
@@ -63,8 +63,21 @@ configure_file("${CMAKE_SOURCE_DIR}/cmake/templates/license.h.in" include/licens
 
 include(EthOptions)
 configure_project(TESTS)
-
+set(LATEST_Z3_VERSION "4.8.12")
 find_package(Z3 4.8.0)
+# NOTE: With EMSCRIPTEN=ON we do not build code from test/ dir and that
+# code is our only reason to require latest Z3.
+if (TESTS AND NOT EMSCRIPTEN)
+  if (${Z3_FOUND})
+    if ( NOT ( "${Z3_VERSION_STRING}" VERSION_EQUAL ${LATEST_Z3_VERSION} ) )
```

```suggestion
```suggestion
  if (
    TESTS AND
    NOT EMSCRIPTEN AND
    ${Z3_FOUND} AND
    NOT ("${Z3_VERSION_STRING}" VERSION_EQUAL ${LATEST_Z3_VERSION})
  )
```

---

Also, some variables are quoted, some are not.

---

It seems Z3_VERSION and Z3_VERSION_STRING can be null even when z3 is found. so I quoted this variable to be more robust against any upcoming changes in z3's find module.
Other variables cannot be null.